### PR TITLE
Add private dns support

### PR DESCRIPTION
**Pull request by BlakeWills**
This PR closes #

The changes in this PR are as follows:

* Adds support for private dns zones

I have read the [contributing guidelines](CONTRIBUTING.md) and have completed the following:

* [x] **Tested my code** end-to-end against a live Azure subscription.
* [x] **Updated the documentation** in the docs folder for the affected changes.
* [x] **Written unit tests** against the modified code that I have made.
* [x] **Updated the [release notes](RELEASE_NOTES.md)** with a new entry for this PR.
* [x] **Checked the coding standards** outlined in the [contributions guide](CONTRIBUTING.md) and ensured my code adheres to them.

If I haven't completed any of the tasks above, I include the reasons why here:

Below is a minimal example configuration that includes the new features, which can be used to deploy to Azure:

```fsharp
```

### CODE REVIEW

File: src/Farmer/Arm/Dns.fs [hunk 1]
```diff
@@ -51,25 +83,36 @@ module DnsRecords =
             .Map(fun r -> r + ".NSRecords")
             .WithOwner(sourceZoneResId)
 
+    let private ttlKey zoneType = match zoneType with | Public -> "TTL" | Private -> "ttl"
```
Comment (BlakeWills) [inline src/Farmer/Arm/Dns.fs:86]: I wonder if I can get rid of this by upgrading the schema for the public records?

Comment (TheRSP) [reply src/Farmer/Arm/Dns.fs:86]: worth trying

File: src/Tests/Dns.fs [hunk 2]
```diff
@@ -306,4 +311,122 @@ let tests = testList "DNS Zone" [
...
+                add_resources [
+                    dnsZone {
+                        name "farmer.com"
+                        zone_type Private
+                        add_records [
+                            cnameRecord {
+                                name "www"
+                                ttl 3600
+                                cname "farmer.com"
+                                zone_type Private
```
Comment (TheRSP) [inline src/Tests/Dns.fs:326]: when adding a record to a zone with `add_records`, should the zone type be automatically set to that of the zone?
I presume it's not possible to add a public record to a private zone or vice versa

**Review by TheRSP — APPROVED** [review]

File: src/Farmer/Arm/Dns.fs (hunk 1 above)
Comment (BlakeWills) [reply src/Farmer/Arm/Dns.fs:86]: Unfortunately we're already using the latest schema for public dns records.

File: src/Farmer/Farmer.fsproj [hunk 3]
```diff
@@ -26,6 +26,7 @@
     <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
     <RepositoryType>git</RepositoryType>
     <RepositoryUrl>https://github.com/CompositionalIT/farmer</RepositoryUrl>
+	  <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
```
Comment (BlakeWills) [inline src/Farmer/Farmer.fsproj:29]: remove

**Review by TheRSP — APPROVED** [review]

**Pull request merged**